hphp/runtime/vm/jit/unique-stubs.cpp

```diff
@@ -1034,12 +1034,15 @@ TCA emitDecRefGeneric(CodeBlock& cb, DataBlock& data) {
   CGMeta meta;
 
   auto const start = vwrap(cb, data, meta, [] (Vout& v) {
+    v << vregrestrict{};
     v << stublogue{};
 
     auto const rdata = rarg(0);
     auto const rtype = rarg(1);
 
     auto const destroy = [&] (Vout& v) {
+      v << vregallow{};
```

Comment:
I really didn't like the idea. I felt like the verbose current design was a little more readable. I gave it a shot though, and the result is pretty clean and quite understandable with the addition of the comments.